HideStatusReason.js

```diff
@@ -0,0 +1,35 @@
+function OnFormLoad(executionContext) {
+    var formContext = executionContext.getFormContext();
+    formContext.data.process.addOnStageChange(function () {
+        HideStatusReasonValue(formContext)
+    });
+}
+
+function HideStatusReasonValue(formContext) {
+    formContext.getControl("statuscode").removeOption(2);
+    formContext.getControl("statuscode").removeOption(970300002);
+    formContext.getControl("statuscode").removeOption(1);
+    formContext.getControl("statuscode").removeOption(970300000); 
+    formContext.getControl("statuscode").removeOption(970300001);         
+
+    var activeStage = formContext.data.process.getActiveStage();
+    var stageName = activeStage.getName();
+
+    if ((formContext.getAttribute("statuscode").getSelectedOption() != null) && (stageName != null)) {
```

Comment:
please move the parts of if clause after getSelectedOption() call on the next line